internal/infraenv/infraenv_test.go

```diff
@@ -109,14 +109,19 @@ var _ = Describe("Delete inactive infraenvs", func() {
 		infraEnv = registerInfraEnv("")
 	})
 
+	// avoid races on slow/fast systems
+	nowPlus5sec := func() strfmt.DateTime {
+		return strfmt.DateTime(time.Now().Add(5 * time.Second))
+	}
+
 	It("Deregister inactive infraEnv", func() {
-		Expect(state.DeleteOrphanInfraEnvs(ctx, 10, strfmt.DateTime(time.Now()))).ShouldNot(HaveOccurred())
+		Expect(state.DeleteOrphanInfraEnvs(ctx, 10, nowPlus5sec())).ShouldNot(HaveOccurred())
```

Comment:
it's not waiting for anything. 
the call to the DB is sync. the race that i suspect  is that the infra env is created and the check is running very fast so the time didn't manage to "catch it" 
So i'm making sure that garbage collector function will check an object from a future perspective.

Reply:
I'm not 100% sure because i didn't managed to reproduce so i cannot add logs.
BeforeEach creates an infra env looking at the time db set the creation time to t1
then the test is running the GC function and checking from garbage infa-envs and use time.Now as a starting point, lets call it t2
In a fast machine the difference between t1 and t2 is probably close enough so that the time check don't see any difference.
To resolve it i'm making sure that t2 will point to at least t1 + 5 sec.

Reply:
ack. I missed what the fix actually does
sounds good